src/flows/pipes/QueryBuilder/CardList.tsx

```diff
@@ -35,6 +35,12 @@ interface Props {
   idx: number
 }
 
+const joinAndDedupeArrays = (arr1: string[], arr2: string[]): string[] => {
+  const deduped = new Set([...arr1, ...arr2])
```

Comment:
this doesn't seem to be setting the correct order. data explorer is floating the selected, but not available in results, items to the top. this seems to be random?